refactor(tool-engine): Optimize tool engine response handling

MLeft opened the pull request:
- Removed the unnecessary "tool response:" prefix from JSON and default responses to prevent confusion for LLMs and reduce token waste.
resolve #13566 

# Summary

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

> [!Tip]
> Close issue syntax: `Fixes #13566 ` , see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.


# Screenshots

| Before | After |
|--------|-------|
| ...    | ...   |

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

Commit:
refactor(tool-engine): Optimize tool engine response handling

- Removed the unnecessary "tool response:" prefix from JSON and default responses to prevent confusion for LLMs and reduce token waste.

crazywoola commented:
Thanks for the PR, please open an issue or link an existing issue in the description first.

MLeft commented:
> Thanks for the PR, please open an issue or link an existing issue in the description first.

It has been associated with #13566

crazywoola commented:
Please fix the lint error by `./dev/reformat`

Commit:
refactor(api): Optimize code format and readability
- In the tool_engine.py file, the code for handling JSON messages has been formatted.
- A more readable multi - line format is used to call the json.dumps function.
- The original logic and functionality are retained, and only the appearance and structure of the code are improved.

crazywoola committed:
fix: ruff lint

wsbao commented:
For version 1.4.1, this issue persists.
